tests/strategies/future/test_future_strategy.py

```diff
@@ -0,0 +1,165 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+import uuid
+from typing import TYPE_CHECKING
+from unittest.mock import Mock, MagicMock
+
+import pytest
+from gsy_framework.constants_limits import GlobalConfig
+from pendulum import today, duration
+
+from gsy_e.constants import TIME_ZONE, FutureTemplateStrategiesConstants
+from gsy_e.models.market.future import FutureMarkets
+from gsy_e.models.strategy.future.strategy import FutureMarketStrategy
+from gsy_e.models.strategy.load_hours import LoadHoursStrategy
+from gsy_e.models.strategy.pv import PVStrategy
+from gsy_e.models.strategy.storage import StorageStrategy
+
+if TYPE_CHECKING:
+    from d3a.models.strategy import BaseStrategy
+
+
+class TestFutureMarketStrategy:
+    """Test the FutureMarketStrategy class."""
+    # pylint: disable = attribute-defined-outside-init
+
+    def setup_method(self) -> None:
+        """Preparation for the tests execution"""
+        GlobalConfig.FUTURE_MARKET_DURATION_HOURS = 24
+        self.time_slot = today(tz=TIME_ZONE).at(hour=12, minute=0, second=0)
+        self.area_mock = Mock()
+        self.area_mock.name = "test_name"
+        self.area_mock.uuid = str(uuid.uuid4())
+        self.future_markets = MagicMock(spec=FutureMarkets)
+        self.future_markets.market_time_slots = [self.time_slot]
+        self.future_markets.id = str(uuid.uuid4())
+
+    @staticmethod
+    def teardown_method() -> None:
+        """Test cleanup"""
+        GlobalConfig.FUTURE_MARKET_DURATION_HOURS = 0
+
+    def _setup_strategy_fixture(self, future_strategy_fixture: "BaseStrategy") -> None:
+        future_strategy_fixture.owner = self.area_mock
+        future_strategy_fixture.update_bid_rates = Mock()
+        future_strategy_fixture.update_offer_rates = Mock()
+        future_strategy_fixture.area = Mock()
+        future_strategy_fixture.area.future_markets = self.future_markets
+        future_strategy_fixture.area.current_tick = 0
+        future_strategy_fixture.area.config = Mock()
+        future_strategy_fixture.area.config.tick_length = duration(seconds=15)
+
+    def test_event_market_cycle_posts_bids_load(self) -> None:
+        """Validate that market cycle event posts bids and offers for the load strategy"""
+        future_strategy_fixture = LoadHoursStrategy(100)
```

Comment:
The name is a bit misleading, It could mean that this instance is a `FutureMarketStrategy` instance when it's not; And reviewing becomes confusing when we have `future_strategy_fixture` and `future_strategy`

Reply:
Indeed, therefore I would go with the rename:
future_strategy_fixture -> load_Strategy_fixture (and pv_strategy_fixture in other tests)